### PR TITLE
Implement SphericalMesh.get_indices_at_coords

### PR DESCRIPTION
**Pull request by EdenRochmanSharabi**
## Summary

Implements `SphericalMesh.get_indices_at_coords`, resolving the remaining work
from #3867 (`RectilinearMesh` was already done in #3824).

The method accepts Cartesian (x, y, z) coordinates, converts to spherical
(r, theta, phi) relative to `self.origin`, validates against grid bounds, and
returns bin indices via `np.searchsorted`. Follows the same pattern as
`CylindricalMesh.get_indices_at_coords`.

Handles the degenerate case where r=0 (theta and phi are mathematically
undefined) by defaulting both to 0.0.

## Changes

- `openmc/mesh.py`: Replace `NotImplementedError` stub with full implementation
- `tests/unit_tests/test_mesh.py`: Add `test_SphericalMesh_get_indices_at_coords`

## Tests

- Basic single-bin lookups along each axis
- Multi-bin angular grids (theta and phi bin assignment)
- Non-default origin
- Boundary values (point on grid edge)
- Degenerate r=0 case
- Out-of-bounds ValueError for r, theta, and phi
- Diagonal point verifying all three coordinates computed correctly

### CODE REVIEW

File: openmc/mesh.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from abc import ABC, abstractmethod
 from collections.abc import Iterable, Sequence, Mapping
 from functools import wraps
-from math import pi, sqrt, atan2
+from math import acos, atan2, pi, sqrt
```
Comment (pshriwise) [inline openmc/mesh.py]: My preference would be to import `math` alone and access these values/functions including the math prefix. Common functions like this can come from a number of places, so it's nice to keep that module context closer to where it's used. These are also names that could be shadowed by local variables and we want to watch out for that as well.

**Review by pshriwise — DISMISSED** [review]
One tiny ask from me here but then I'd say this is ready @EdenRochmanSharabi. Thank you!

**Comment by pshriwise** [comment]
It looks like this might also need to be updated against the `develop` branch to use some of the updated action versions from #3913.

**Review by paulromano — APPROVED** [review]
